### PR TITLE
issue #SB-15929 fix: changes for user framework update making all fil…

### PR DESCRIPTION
**Pull request by manzarul**
* changes for user framework update. making all framework attribute as optional. value and type will be verified , once user provide it, in case of null or empty list it will be ignored.

### CODE REVIEW

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-utils/sunbird-utils/pullRequest?prid=4558434&bid=15331434)

File: sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql [hunk 1]
```diff
@@ -0,0 +1 @@
+update system_settings set value ='{"fields":["firstName","lastName","profileSummary","avatar","countryCode","dob","email","gender","grade","language","location","phone","subject","userName","webPages","jobProfile","address","education","skills","badgeAssertions"],"publicFields":["firstName","lastName","profileSummary","userName"],"privateFields":["email","phone"],"csv":{"supportedColumns":{"NAME":"firstName","MOBILE PHONE":"phone","EMAIL":"email","SCHOOL ID":"orgId","USER_TYPE":"userType","ROLES":"roles","USER ID":"userId","SCHOOL EXTERNAL ID":"orgExternalId"},"outputColumns":{"userId":"USER ID","firstName":"NAME","phone":"MOBILE PHONE","email":"EMAIL","orgId":"SCHOOL ID","orgName":"SCHOOL NAME","userType":"USER_TYPE","orgExternalId":"SCHOOL EXTERNAL ID"},"outputColumnsOrder":["userId","firstName","phone","email","organisationId","orgName","userType","orgExternalId"],"mandatoryColumns":["firstName","userType","roles"]},"read":{"excludedFields":["avatar","jobProfile","address","education","webPages","skills"]},"framework":{"fields":["board","gradeLevel","medium","subject","id"]}}' where id='userProfileConfig';
```
Comment (anmol2302) [inline sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql]: @manzarul mandatory field is missing from fields

Comment (anmol2302) [inline sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql]: @manzarul please add mandatory fields in framework

Comment (anmol2302) [inline sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql]: @manzarul please change system_settings to sunbird.system_settings.

File: sunbird-platform-core/common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 2]
```diff
@@ -1117,4 +1118,116 @@ public void testValidateVerifyUserFailureWithEmptyId() {
...
+              request.getRequest(), getSupportedFileds(), frameworkMandatoryFields);
+      response = true;
+    } catch (Exception e) {
+      Assert.assertTrue(response);
+    }
+    Assert.assertTrue(response);
+  }
+
+  @Test
+  public void testValidateMandatoryFrameworkFiledValueAsString() {
```
Comment (anmol2302) [inline sunbird-platform-core/common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: typo in method name

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-utils/sunbird-utils/pullRequest?prid=4558434&bid=15331434)

File: sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql (hunk 1 above)
Comment (manzarul) [reply sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql]: Done

Comment (manzarul) [reply sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.89_cassandra.cql]: done

File: sunbird-platform-core/common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java (hunk 2 above)
Comment (manzarul) [reply sunbird-platform-core/common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: one

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-utils/sunbird-utils/pullRequest?prid=4558434&bid=15331434)

**Pull request merged**